src/charm.py

```diff
@@ -69,27 +69,84 @@ def _relation_created(self, relation_name: str) -> bool:
 
     def _on_collect_status(self, event: CollectStatusEvent) -> None:
         """Collect status for the charm."""
-        if not self._private_key_is_stored:
-            event.add_status(WaitingStatus("Waiting for private key to be generated"))
-            return
-        if not self._csr_is_stored:
-            event.add_status(WaitingStatus("Waiting for CSR to be generated"))
+        if not self._mode_config_is_valid():
+            event.add_status(
+                BlockedStatus("Invalid mode configuration: only 'unit' and 'app' are allowed")
+            )
             return
-        if not self._certificates_relation_created:
-            event.add_status(ActiveStatus())
+        mode = self._get_config_mode()
+        if mode == "unit":
+            status = self._collect_unit_status()
+            event.add_status(status)
             return
-        if not self._certificate_is_stored:
-            event.add_status(ActiveStatus("Certificate request is sent"))
+        elif mode == "app":
+            status = self._collect_app_status()
+            event.add_status(status)
             return
-        event.add_status(ActiveStatus("Certificate is available"))
+
+    def _collect_unit_status(self) -> StatusBase:
```

Comment:
We may want to reconsider the names of these two functions, since they are very similar to `on.collect_unit_status` and `on.collect_app_status` from the ops framework, but have different meanings. It may be misleading.

Maybe `_collect_status_unit_mode` to conform with the `_configure` variants

Reply:
Done